amr_kitchen/plotfile_cooker.py

```diff
@@ -203,17 +203,55 @@ def __init__(self, fields, boxes, field_arg, limit_level):
             raise IndexError((f"The field indexing [{field_arg}] is not"
                               f" compatible with the number of fields"
                               f" in the plotfile ({len(fields)})"))
-        self.boxes = boxes
+        self.cells = cells
         self.fields = fields
         self.limit_level = limit_level
+        self.boxes = boxes
 
     def __getitem__(self, key):
         if key > self.limit_level:
             raise ValueError((f"The maximum AMR level of the plotfile"
                               f" is {self.limit_level}"))
-        return LevelDataStream(self.boxes[key]['files'],
-                               self.boxes[key]['offsets'],
+        return LevelDataStream(self.cells[key]['files'],
+                               self.cells[key]['offsets'],
                                self.farg)
+    
+    def __call__(self, cord_x, cord_y, cord_z = None):
+        if cord_z == None:
+            key = (cord_x, cord_y)
+        else:
+            key = (cord_x, cord_y, cord_z)
+
+        # Input validation
+        if self.boxes == None:
+            raise KeyError("PlotfileCooker object was not initiated with a True 'point' argument")
+        if len(key) != self.limit_level+1:
+            raise KeyError(f"Desired point {key} of {len(key)}d size doesn't correlate with the {self.limit_level+1}d size of the plotfile")
+
+        # Let's see in which box does the point belong 
+        matching_indicies = []
+        for box in range(len(self.boxes[self.limit_level])):
+            sous_box = self.boxes[self.limit_level][box]
+            for component in range(len(key)):
+                if key[component] < sous_box[component][0] or key[component] > sous_box[component][-1]:
+                    break
+            else:
+                matching_indicies.append(box)
+
+        # More validation and bug catching
+        if len(matching_indicies) == 0:
+            raise KeyError(f"An unexpected error occured.\n Chosen point {key} not related to any box in the plotfile")
+        # If the point is in only one box 
+        elif len(matching_indicies) == 1:
+            return self[self.limit_level][matching_indicies[0]]
+        # If the point is between more than one box
+        else:
+            conc_array = self[self.limit_level][matching_indicies[0]]
+            for index in range(1, len(matching_indicies)):
+                # The data of all boxes that contain the point gets concatenated 
+                conc_array = np.concatenate((conc_array, self[self.limit_level][matching_indicies[index]]))
```

Comment:
Some code to perform the interpolation in a single matching box
```python
# box_matches = boolean array at each level with matching boxes    
# Finest matching level (because finer levels are covered by coarser levels)
    match_level = len(box_matches) - 1
    # id of the matching box
    box_id = np.nonzero(box_matches[-1])[0]
    # Global indices of the box
    box_indices = pck.cells[match_level]['indexes'][box_id[0]]
    # 3D data for a single box at the finest matching level
    data = pck[field][match_level][box_id][0]
    # Convert point to box indices
    # Global point indices (double check this if the boundary are well handled (x = 0 -> idx = 0))
    point_indices = (point / (pck.geo_high - np.array(pck.geo_low))) * pck.grid_sizes[match_level]
    # point indices local to box 
    point_indices -= box_indices[0]
    # Compute the point value
    point_data = map_coordinates(data, np.transpose([point_indices]))
    #return point_data
```